src/App.tsx

```diff
@@ -1,44 +1,97 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
-
 import classNames from 'classnames';
+
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { getUserPosts, getUsers } from './utils/fetchClient';
+import { IUser } from './types/User';
+import { IPost } from './types/Post';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<IUser[] | null>(null);
+  const [activeUser, setActiveUser] = useState<IUser | null>(null);
+
+  const [posts, setPosts] = useState<IPost[] | null>(null);
+  const [activePost, setActivePost] = useState<IPost | null>(null);
+
+  const [isError, setIsError] = useState<boolean>(false);
+  const [isLoding, setIsLoading] = useState<boolean>(false);
+
+  const changeUser = useCallback((user: IUser) => {
+    setActiveUser(user);
+  }, []);
```

Comment:
```suggestion
  const changeUser = useCallback((user: IUser) => {
    setActiveUser(user);
  }, [user]);
```
You need to add all the arguments you pass to your function to the dependency array to prevent unnecessary renders. Please fix all cases
https://reactjs.org/docs/hooks-reference.html#usecallback

Reply:
i can't pass user to dependency array. Because I use this function in another component. I think that in this case the mass of dependencies can be empty. Because we need to take the value and add it to the state.